scripts/httpd/upload.cgi

```diff
@@ -122,11 +122,54 @@ def ensure_namespaced(name, namespace):
     return name
 
 
-def main():
-    form = cgi.FieldStorage()
+def get_checksum_and_hash_type(form):
+    # Search for the file hash, start with stronger hash functions
+    if 'sha512sum' in form:
+        checksum = check_form(form, 'sha512sum')
+        hash_type = "sha512"
+
+    elif 'md5sum' in form:
+        # Fallback on md5, as it's what we currently use
+        checksum = check_form(form, 'md5sum')
+        hash_type = "md5"
+
+    else:
+        send_error('Required checksum is not present',
+                   status='400 Bad Request')
+
+    return checksum, hash_type
+
+
+def emit_fedmsg(config, name, checksum, filename, username, msgpath):
+    # Emit a fedmsg message.  Load the config to talk to the fedmsg-relay.
+    if config.getboolean('upload', 'fedmsgs', fallback=True):
+        try:
+            import fedmsg
+            import fedmsg.config
+
+            config = fedmsg.config.load_config([], None)
+            config['active'] = True
+            config['endpoints']['relay_inbound'] = config['relay_inbound']
+            fedmsg.init(name="relay_inbound", cert_prefix="lookaside", **config)
+
+            topic = "lookaside.new"
+            msg = dict(name=name, md5sum=checksum,
```

Comment:
I wonder what would happen if actually fix to say just "checksum". Anyway this is for another patch.